float64[6] positions
float64[6] velocities